cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -1727,29 +1728,28 @@ pSigningKeyFile fdir =
 
 pKesPeriod :: Parser KESPeriod
 pKesPeriod =
-  KESPeriod <$>
-    Opt.option Opt.auto
-      (  Opt.long "kes-period"
-      <> Opt.metavar "NATURAL"
-      <> Opt.help "The start of the KES key validity period."
-      )
+  fmap KESPeriod $ Opt.option (Opt.bounded "KES_PERIOD") $ mconcat
+    [ Opt.long "kes-period"
+    , Opt.metavar "KES_PERIOD"
+    , Opt.help "The start of the KES key validity period."
+    ]
 
 pEpochNo :: Parser EpochNo
 pEpochNo =
   EpochNo <$>
-    Opt.option Opt.auto
+    Opt.option (Opt.bounded "EPOCH")
       (  Opt.long "epoch"
-      <> Opt.metavar "NATURAL"
+      <> Opt.metavar "EPOCH"
```

Comment:
So I would argue to leave `NATURAL` so users know what kind of value they need to enter and can therefore infer the bounds. The help already says `"The epoch number."` so I don't see the additional value putting `"EPOCH"` as the metavar brings. I don't have a strong opinion but I'm interested to know why you made the change.

Reply:
I made the change because if there is a CLI parser error it will say:

```
option --invalid-before: EPOCH must not be less than 0
```

so I changed the `metavar` to match.  Let me know what you think.

Reply:
Fair :+1: